pkgs/unified_analytics/lib/src/utils.dart

```diff
@@ -117,14 +117,14 @@ bool legacyOptOut({
       if (dartObj.containsKey('enabled') && dartObj['enabled'] == false) {
         return true;
       }
+    } on FormatException {
```

Comment:
should catch FileSystemException too for `readAsStringSync()`.    I think those are the only two exceptions so I think it's OK to drop the catch all.